### ISSUE TITLE
Mismatch between query and query mocks

### ISSUE THREAD

**Issue by ChristianOezelt** [apollographql/apollo-test-utils#24]
Hi,

I am facing an issue where the mocked network interface can not map sent queries to the defined mocks. This is due to Apollo changing the queries, but not the queries provided in the mocks.

Please note that I only use the mockNetworkInterface without the MockedProvider because the MockedProvider does not support features I need, which is why I wrote my own TestProvider, which basically just uses the ApolloProvider with said mocked network interface 

For both, the actual query that is sent and the query I provide in the mock, I use the same query which is defined in a separate file:

#### IntrospectionQuery.js
```javascript
import { gql } from 'react-apollo'

export const introspectionQuery = gql`query( $typeName: String! ) {
  __type( name: $typeName ){
    name
    inputFields{
      name
      type{
        name
        kind
        ofType{
          name
          kind
        }
      }
    }
  }
}`;
```
When providing mocks to the mocked network interface, it stringifies the query with its variables and uses it in an object as an key, mapping the provided result to it.

The issue now is that standard Apollo is changing the query, resulting in a mismatch between the sent query and the queries in the mock.

#### The stringified sent query looks like this:
```
{  
   "variables":{  
      "typeName":"TestType"
   },
   "query":"query ($typeName: String!) {\n  __type(name: $typeName) {\n    name\n    inputFields {\n      name\n      type {\n        name\n        kind\n        ofType {\n          name\n          kind\n          \n        }\n        __typename\n      }\n      __typename\n    }\n    __typename\n  }\n}\n"
}
```
#### The mock I defined for this query
```
{  
   "variables":{  
      "typeName":"TestType"
   },
   "query":"query ($typeName: String!) {\n  __type(name: $typeName) {\n    name\n    inputFields {\n      name\n      type {\n        name\n        kind\n        ofType {\n          name\n          kind\n        }\n      }\n    }\n  }\n}\n"
}
```

In other words the query which the mock provider is looking for looks like this:
#### Formated query
```
query ($typeName: String!) {
  __type(name: $typeName) {
    name
    inputFields {
        name
        type {
            name
            kind
            ofType {
                name
                kind
            }
            __typename
        }
        __typename
    }
    __typename
    }
}
```
having the __typename added, the stringified queries match no more and my testcases fail.

**Comment by helfer** [comment]
Hi @ChristianOezelt, yeah, this is an issue we run into as well with our tests. I think the best solution would be to take the `addTypename` from apollo-client and add it as an option to the mock interface. Would you by any chance have time to make a PR? :pray:

**Comment by ChristianOezelt** [comment]
I'll probably have a look at it on the weekend, can't yet promise anything
